src/main/java/uk/gov/hmcts/reform/logging/exception/AbstractLoggingException.java

```diff
@@ -15,4 +22,28 @@ protected AbstractLoggingException(String message) {
     protected AbstractLoggingException(String message, Throwable cause) {
         super(message, cause);
     }
+
+    private static void triggerBadImplementationLog(Throwable cause) {
+        Throwable invalid = new InvalidExceptionImplementation("AlertLevel is mandatory as per configuration", cause);
+
+        log.error("Bad implementation of '" + cause.getClass().getCanonicalName() + "' in use", invalid);
+    }
+
+    public static AbstractLoggingException getFromLogEvent(ILoggingEvent event) {
+        Throwable eventException = ((ThrowableProxy) event.getThrowableProxy()).getThrowable();
+
+        try {
+            return (AbstractLoggingException) eventException;
+        } catch (ClassCastException e1) {
+            // for spring boot projects there's a generic exception wrapper
+            // let's try to cast the cause instead
+            try {
+                return (AbstractLoggingException) eventException.getCause();
```

Comment:
is it better to do an instanceof check rather than nesting try...catch?

Reply:
my first choice was `instanceof`, but it didn't work